packages/apollos-ui-connected/src/NodeSingleConnected/useInteractWithNode.js

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from 'react';
+import { gql, useMutation } from '@apollo/client';
+import isObject from 'lodash/isObject';
+
+export const INTERACT_WITH_NODE = gql`
+  mutation interactWithNode(
+    $nodeId: ID!
+    $action: InteractionAction!
+    $data: [InteractionDataField]
+  ) {
+    interactWithNode(nodeId: $nodeId, action: $action, data: $data) {
+      success
+    }
+  }
+`;
+
+function transformVariables(variables) {
+  if (!isObject(variables?.data)) {
+    return variables;
+  }
+
+  // Map the `data` object to an array of key/value pairs,
+  // to conform to InteractionDataField schema
+  const transformedData = Object.entries(variables?.data).map(
+    ([field, value]) => ({
+      field,
+      value,
+    })
+  );
+  const nodeId = variables.videoId;
+  const { action } = variables;
+  return {
+    nodeId,
+    action,
+    data: transformedData,
+  };
+}
+
+function useInteractWithNode(options = {}) {
+  const [_mutation, result] = useMutation(INTERACT_WITH_NODE, options);
+
+  /*
+  Create a wrapped mutation function that:
+    - Enforces authentication
+    - Transforms the input variables, so hook consumers don't have to worry
+      about conforming to Apollos API schema for [InteractionDataField]
+  */
+  const mutation = useCallback(
+    (mutOpts = { variables: {} }) => {
+      return _mutation({
+        ...mutOpts,
+        variables: transformVariables(mutOpts.variables),
+      });
+    },
+    [_mutation]
```

Comment:
Should options be listed here, just to be safe? (It might not be necessary since the reference to `_mutation` will likely change if `options` changes.) 

```suggestion
    [_mutation, options]
```